### PR TITLE
Improve this complicated script by using the proper enterprise patterns

### PR DESCRIPTION
**Pull request by december0123**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,7 +1,38 @@
 #!/usr/bin/env python3
```
Comment (kmachnicki) [inline main.py]: There should be an empty line after a shebang to further improve the readability of the script.

File: main.py [hunk 2]
```diff
@@ -1,7 +1,38 @@
 #!/usr/bin/env python3
+# This is the Hello-World-Python program ported to Python3 and improved
+# by using the proper enterprise project patterns.
+# Originally written by kmachnicki
+# Disclaimer: Yes, it is a joke :)
 
-#Welcome!
```
Comment (kmachnicki) [inline main.py]: Unnecessary empty line.

File: main.py [hunk 3]
```diff
@@ -1,7 +1,38 @@
...
+# Originally written by kmachnicki
+# Disclaimer: Yes, it is a joke :)
 
-#Welcome!
 
-print("Hello World!")
+# This is the heart of the program.
+# This class is responsible for printing the string provided by the caller.
+# It is a proxy for the print function.
+class Printer:
```
Comment (kmachnicki) [inline main.py]: Unit tests are needed for the Printer class to fully check its capabilities.

File: main.py (hunk 1 above)
Comment (kmachnicki) [inline main.py]: According to PEP 263 the encoding line is recommended after a shebang, like this:
`# -*- coding: utf8 -*-`

File: main.py [hunk 4]
```diff
@@ -1,7 +1,38 @@
...
+    # __init__ is responsible for initializing the _printer member, so we can use it later.
+    def __init__(self):
+        self._printer = print
 
-#And that's it ;)
+    # __call__ is used to print the string provided by the caller
+    def __call__(self, string_to_print):
```
Comment (kmachnicki) [inline main.py]: Every function that accepts or returns some arguments, must have the description formatted the following way:

```
Parameters
----------
X : {type}
     Description of the parameter.

Returns
-------
Y : {type}
    Description of returned value.
```

Please apply this to all other functions.

File: main.py [hunk 5]
```diff
@@ -1,7 +1,38 @@
...
+# Disclaimer: Yes, it is a joke :)
 
-#Welcome!
 
-print("Hello World!")
+# This is the heart of the program.
+# This class is responsible for printing the string provided by the caller.
+# It is a proxy for the print function.
+class Printer:
+    # __init__ is responsible for initializing the _printer member, so we can use it later.
```
Comment (kmachnicki) [inline main.py]: Class/Functions/Methods descriptions should happen using the triple quote multiline comment. Please apply to all functions.

**Comment by december0123** [comment]
Hello @kmachnicki. I've stumbled upon your beautiful script and I thought I'd add a little bit to it. Please accept this pull request.

**Review by kmachnicki — COMMENTED** [review]
Please comply to comments.

**Comment by kmachnicki** [comment]
Thank you for improving the contents of this script. Your hard work is much appreciated.

**Review by kmachnicki — APPROVED** [review]
Nicely done.

**Pull request merged**